Dockerfile

```diff
@@ -1,10 +1,38 @@
-FROM ruby:2.5.1
-RUN apt-get update -qq && apt-get install -y ruby-dev
+# Install gems
+FROM alpine:3.8 as build
+
+RUN \
+apk --no-cache add build-base ruby-dev ruby-bundler git openssl-dev && \
+echo 'gem: --no-document' > /etc/gemrc && \
+bundle config --global silence_root_warning 1
 
 RUN mkdir /bolt-server
-WORKDIR /bolt-server
+# Gemfile requires gemspec which requires bolt/version which requires bolt
 ADD . /bolt-server
-RUN bundle install
+WORKDIR /bolt-server
+RUN bundle install --no-cache --path vendor/bundle
+
+# Final image
+FROM alpine:3.8
+ARG bolt_version=no-version
+LABEL org.label-schema.maintainer="Puppet Bolt Team <team-direct-change-bolt@puppet.com>" \
+      org.label-schema.vendor="Puppet" \
+      org.label-schema.url="https://github.com/puppetlabs/bolt" \
+      org.label-schema.name="PE Bolt Server" \
+      org.label-schema.license="Apache-2.0" \
+      org.label-schema.version=${bolt_version} \
+      org.label-schema.vcs-url="https://github.com/puppetlabs/bolt" \
+      # Same with these
+      #org.label-schema.vcs-ref="b75674e1fbf52f7821f7900ab22a19f1a10cafdb" \
+      #org.label-schema.build-date="2018-05-09T20:10:01Z" \
+      #org.label-schema.schema-version="1.0" \
+      org.label-schema.dockerfile="/Dockerfile"
+
+RUN \
+apk --no-cache add ruby-dev openssl-dev ruby-bundler
```

Comment:
I don't think we need the `dev` versions of ruby or openssl here.

Reply:
It didn't work with regular SSL - puma would complain that SSL bindings weren't available :woman_shrugging:

Reply:
Let me know if it works with ssl on your machine and I can investigate further.